plugins/hls-alternate-number-format-plugin/src/Ide/Plugin/Literals.hs

```diff
@@ -0,0 +1,401 @@
+{-# LANGUAGE DeriveGeneric     #-}
```

Comment:
I think this whole module can be replaced by a fitting call to [`syb`](https://hackage.haskell.org/package/syb).

Reply:
Shout if you require assistance here

Reply:
So just taking a look briefly at syb and the (very slim documentation, I found one paper related to it). The idea is to generify (is that word) each `traverseLBlah` into just a few functions that unwrap data constructors into it's components and recurses until `LHsExpr` is encountered?

Reply:
Basically yes, in your particular issue, I think you can get away with only matching on the leaf constructors (e.g. your literal constructors).  I *think* `everywhereM` is useful for you.